### PR TITLE
Updated linear and circular progressbar topics

### PR DESCRIPTION
**Pull request by ddincheva**

### CODE REVIEW

File: en/components/circular_progress.md [hunk 1]
```diff
@@ -9,19 +9,107 @@ _keywords: Ignite UI for Angular, UI controls, Angular widgets, web widgets, UI
...
+
+@NgModule({
+    ...
+    imports: [..., IgxProgressBarModule],
+    ...
+})
+export class AppModule {}
+```
+And now to have a better understanding how everything works, let's create a simple example, in which we will simulate a real process progress, that is
+triggered on button click. In order to make out example better we will need to import and some additional modules in the **app.module.ts** file.
```
Comment (ig-ealbert) [inline en/components/circular_progress.md]: In order to make *our* example better

Comment (ig-ealbert) [inline en/components/circular_progress.md]: we will need to import some additional modules

File: en/components/circular_progress.md [hunk 2]
```diff
@@ -9,19 +9,107 @@ _keywords: Ignite UI for Angular, UI controls, Angular widgets, web widgets, UI
...
+    ..., IgxButtonModule, IgxIconModule, IgxRippleModule
+} from 'igniteui-angular/main';
+
+@NgModule({
+    ...
+    imports: [..., IgxButtonModule, IgxIconModule, IgxRippleModule],
+})
+export class AppModule {}
+```
+Notice that the **igx-circular-bar** emits `onProgressChanged` event that outputs object that give us `{currentValue: 65, previousValue: 64}` on each step.
```
Comment (ig-ealbert) [inline en/components/circular_progress.md]: that outputs an object that gives us

File: en/components/circular_progress.md [hunk 3]
```diff
@@ -9,19 +9,107 @@ _keywords: Ignite UI for Angular, UI controls, Angular widgets, web widgets, UI
...
+       this.interval = clearInterval(this.interval);
+     }
+  }
+  public progresChanged(progress) { ... }
+  private randomIntFromInterval(min: number, max: number) {
+    return Math.floor(Math.random() * (max - min + 1) + min);
+  }
+```
+
+And now if we set up everything correct we should have the folllowing displayed in our browser:
```
Comment (ig-ealbert) [inline en/components/circular_progress.md]: if we set up everything correctly
What I wrote is grammatically correct, although enough native English speakers don't know that and would be ok with what you wrote. :)

File: en/components/circular_progress.md [hunk 4]
```diff
@@ -30,6 +118,7 @@ _keywords: Ignite UI for Angular, UI controls, Angular widgets, web widgets, UI
 | `max` |  number | Set maximum value that can be passed. Default `max` value is 100. |
 | `value` |  number | Set value that indicates the completed bar position. |
 | `animate` |  boolean | animation on progress bar. |
+| `textVisibility` | boolean | Set the text to be visible. By default is set to `true`. |
```
Comment (ig-ealbert) [inline en/components/circular_progress.md]: Either "by default it is set to `true`" or "the default value is `true`".

File: en/components/linear_progress.md [hunk 5]
```diff
@@ -9,29 +9,267 @@ _keywords: Ignite UI for Angular, UI controls, Angular widgets, web widgets, UI
...
+@NgModule({
+    ...
+    imports: [..., IgxProgressBarModule],
+    ...
+})
+export class AppModule {}
+```
+
+And now let's create a simple example in which we are going to simulate a continuous process, that is triggered on button click. In addition you can notice
+that the speed of loading depends on the `max` attribute that we are set to our `igx-linear-bar`. In first case our max value is set to 200 and loads quite slower,
```
Comment (ig-ealbert) [inline en/components/linear_progress.md]: that we have set to our `igx-linear-bar` 
or
that we are setting to our `igx-linear-bar`.
In *the* first case, our max value is set to 200 and loads slower than in the second where *it* is set to 100

File: en/components/linear_progress.md [hunk 6]
```diff
@@ -9,29 +9,267 @@ _keywords: Ignite UI for Angular, UI controls, Angular widgets, web widgets, UI
...
+```
+If all went well, you should see something like the following in your browser:
+<div class="sample-container loading" style="height:500px">
+    <iframe id="linear-progressbar-iframe" frameborder="0" seamless width="100%" height="100%" src="{environment:demosBaseUrl}/linear-progressbar" onload="onSampleIframeContentLoaded(this);"></iframe>
+</div>
+<div>
+<button data-localize="stackblitz" class="stackblitz-btn" data-iframe-id="linear-progressbar-iframe" data-demos-base-url="{environment:demosBaseUrl}">view on stackblitz</button>
+</div>
+<div class="divider--half"></div>
+And now let's enhance our example and create a different types of loading bars, that can be striped or not.
```
Comment (ig-ealbert) [inline en/components/linear_progress.md]: and create different types (if more than one type)
and create a different type (if only one type)

File: en/components/linear_progress.md [hunk 7]
```diff
@@ -9,29 +9,267 @@ _keywords: Ignite UI for Angular, UI controls, Angular widgets, web widgets, UI
...
+
+  public reset() {
+    this.linearBars.toArray().forEach((bar) => bar.value = 0);
+  }
+
+  private randomIntFromInterval(min: number, max: number) {
+      return Math.floor(Math.random() * (max - min + 1) + min);
+  }
+```
+So if we set up everything correct, let's see what happened in the browser:
```
Comment (ig-ealbert) [inline en/components/linear_progress.md]: correctly, but again, this isn't a big deal :)

File: en/components/linear_progress.md [hunk 8]
```diff
@@ -9,29 +9,267 @@ _keywords: Ignite UI for Angular, UI controls, Angular widgets, web widgets, UI
...
+<div class="sample-container loading" style="height:550px">
+    <iframe id="linear-sample-1-iframe" frameborder="0" seamless width="100%" height="100%" src="{environment:demosBaseUrl}/linear-progressbar-sample-1" onload="onSampleIframeContentLoaded(this);"></iframe>
+</div>
+<div>
+<button data-localize="stackblitz" class="stackblitz-btn" data-iframe-id="linear-sample-1-iframe" data-demos-base-url="{environment:demosBaseUrl}">view on stackblitz</button>
+</div>
+<div class="divider--half"></div>
+
+Finally let's make our sample even more exciting and good as we set the following attributes: `textAlign`, `textVisibility`, `textTop` and `text`.
+And now let's see how looks like our code:
```
Comment (ig-ealbert) [inline en/components/linear_progress.md]: let's see how our code looks

File: en/components/linear_progress.md [hunk 9]
```diff
@@ -9,29 +9,267 @@ _keywords: Ignite UI for Angular, UI controls, Angular widgets, web widgets, UI
...
+        <igx-linear-bar type="default"></igx-linear-bar>
+        <h5>Text is aligned CENTER</h5>
+        <igx-linear-bar type="info" [textTop]="false" [textAlign]="positionCenter" [striped]="true"></igx-linear-bar>
+        <h5>Text is aligned END</h5>
+        <igx-linear-bar type="success" [textTop]="false" [textAlign]="positionEnd"></igx-linear-bar>
+        <h5> Set your custom text</h5>
+        <igx-linear-bar type="warning" [text]="'Custom text'" [textAlign]="positionCenter" [striped]="true"></igx-linear-bar>
+        <h5>Text is set above the line</h5>
+        <igx-linear-bar type="danger" [textTop]="true"></igx-linear-bar>
+        <h5>Witout text</h5>
```
Comment (ig-ealbert) [inline en/components/linear_progress.md]: Without

File: en/components/linear_progress.md [hunk 10]
```diff
@@ -9,29 +9,267 @@ _keywords: Ignite UI for Angular, UI controls, Angular widgets, web widgets, UI
...
+<div class="button-container">
+    <p>Press the button to reset the bars</p>
+    <button igxButton="fab" igxButtonBackground="#333" igxRipple="white" (click)="reset()" [disabled]="disable">
+        <igx-icon name="replay" color="white" isActive="true" ></igx-icon>
+    </button>
+</div>
+...
+```
+
+And do not forget to firts import `IgxTextAlign` in your component.
```
Comment (ig-ealbert) [inline en/components/linear_progress.md]: first

File: en/components/linear_progress.md [hunk 11]
```diff
@@ -9,29 +9,267 @@ _keywords: Ignite UI for Angular, UI controls, Angular widgets, web widgets, UI
...
 ### API
 | Name   |       Type      |  Description |
 |:----------|:-------------:|:------|
-| `max` |  number | Set maximum value that can be passed. |
+| `max` |  number | Set maximum value that can be passed. By default it is set to 100. |
 | `type` |  string | Set type of the linear bar. Possible options - `default`, `success`, `info`, `warning`, and `danger`. |
 | `value` |  number | Set value that indicates the completed bar position. |
 | `striped` |  boolean | Set bar to have striped style. |
 | `animate` |  boolean | animation on progress bar. |
+| `textAlign` | enum | Set the position that define where the text is aligned. Possible options - `IgxTextAlign.START` (default), `IgxTextAlign.CENTER`, `IgxTextAlign.END`. |
```
Comment (ig-ealbert) [inline en/components/linear_progress.md]: that defines

File: en/components/linear_progress.md [hunk 12]
```diff
@@ -9,29 +9,267 @@ _keywords: Ignite UI for Angular, UI controls, Angular widgets, web widgets, UI
...
+| `textVisibility` | boolean | Set the text to be visible. By default is set to `true`. |
```
Comment (ig-ealbert) [inline en/components/linear_progress.md]: By default it is set to `true`.

File: en/components/linear_progress.md [hunk 13]
```diff
@@ -9,29 +9,267 @@ _keywords: Ignite UI for Angular, UI controls, Angular widgets, web widgets, UI
...
+| `textTop` | boolean | Set the position that defene is text to be aligned above the progress line. By default is set to `false`. |
```
Comment (ig-ealbert) [inline en/components/linear_progress.md]: Set the position that defines if the text should be aligned above the progress line.
Or at least that's what I think you're trying to say?

File: en/components/linear_progress.md [hunk 14]
```diff
@@ -9,29 +9,267 @@ _keywords: Ignite UI for Angular, UI controls, Angular widgets, web widgets, UI
...
+| `text` | string | Set a custom text that is displayed according defined position. |
```
Comment (ig-ealbert) [inline en/components/linear_progress.md]: according to the defined position

File: en/components/circular_progress.md (hunk 3 above)
Comment (zdrawku) [reply en/components/circular_progress.md]: 👍

File: en/components/linear_progress.md (hunk 13 above)
Comment (zdrawku) [reply en/components/linear_progress.md]: Yes, this property is used to define the text-top position

**Pull request merged**